### DISCUSSION TITLE
How can I add SVGs and other HTML elements using remark-directive?

### DISCUSSION THREAD

**Discussion by orkito96** [remarkjs/remark#1098] [Q&A] [answered]
I am trying to create admonitions like [remark-admonitions](https://github.com/elviswolcott/remark-admonitions) using [remark-directive](https://github.com/remarkjs/remark-directive).
As of now I'm only able to generate something like this `<div class="admonition tip"><p>This is a tip admonition with markdown <code>syntax</code> <strong>support</strong>.</p></div>` for
```md
:::tip
This is a tip admonition with markdown `syntax` **support**.
:::
```
This is nothing more than just a few basic things on top of remark-directive's examples. I have tried a lot but still can't figure out how to add SVGs and other HTML elements while using remark-directive. 
Currently I'm using something like this
```ts
/** @type {import('unified').Plugin<[], import('mdast').Root>} */
function myRemarkPlugin() {

// const svgs: { [key: string]: string } = {
//   note: "<svg>...note...</svg>",
//   info: "<svg>...info...</svg>",
//   tip: "<svg>...tip...</svg>",
// };

  return (tree) => {
    visit(tree, (node) => {
      if (
        node.type === 'textDirective' ||
        node.type === 'leafDirective' ||
        node.type === 'containerDirective'
      ) {
        if (node.name !== 'note' && node.name !== 'tip' && node.name !== 'info') return

        const data = node.data || (node.data = {})

        data.hName = "div"
        data.hProperties = {
          className: ["admonition", node.name],
        };
      }
    })
  }
}
```
What I'm trying to have is something like this;
```html
<div class="admonition tip">
  <span class="admonition-heading"><svg>...tip...</svg>TIP</span>
  <p>This is a tip admonition.</p>
</div>
```
What am I missing here and how to add SVGs and other HTML elements with remark while using remark-directive?

Thanks for the help and have a nice day.

**Comment by sudo-subg** [comment, marked as answer]
You can do something like this;
```js
node["children"].unshift({
  type: "html",
  value: `<span class="admonition-heading">${svgs[node.name]}${node.name.toUpperCase()}</span>`,
});
```

**Reply by orkito96** [reply]
Awesome.

**Reply by wooorm** [reply]
If possible, I recommend not to inject strings of HTML: we’re building an AST here, and strings of raw markup defeat a bit of that purpose. Instead, build ASTs, make nodes. You can inject `hChildren` just like you can set `hProperties` and such: https://github.com/syntax-tree/mdast-util-to-hast#fields-on-nodes.
